### PR TITLE
E2e ci localserver

### PR DESCRIPTION
**Pull request by esaminu**
- Adds functionality to bundle code in docker container and serve the code locally to run the e2e tests against.
- Have one bank sub account per worker to act as bank for tests run by that worker. This helps sidestep nonce retries exceeded errors.
- Pass each worker `bankAccount` through a fixture scoped by worker.
- Add functionality to keep track of amount spent by each worker.
- Adds a custom logger that outputs amount spent by each worker, total amount spent and difference in `bankAccount`.
- Integrates Pagerduty for alerts when a test run fails. 
- Extend `JsonRpcProvider` to catch `NotEnoughBalance` errors and use `near-contract-helper` to create a new account and reload on `testnet`.

### CODE REVIEW

File: packages/e2e-tests/utils/SelfReloadingJSONRpcProvider.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const { BN } = require("bn.js");
+const { parseNearAmount } = require("near-api-js/lib/utils/format");
+
+const { createAccountWithHelper } = require("../services/contractHelper");
+const E2eTestAccount = require("./E2eTestAccount");
+const { generateTestAccountId, getWorkerAccountRegex } = require("./helpers");
+const nearApiJsConnection = require("./connectionSingleton");
+const { getTestAccountSeedPhrase } = require("./helpers");
+
+class SelfReloadingJSONRpcProvider extends JsonRpcProvider {
```
Comment (MaximusHaximus) [inline packages/e2e-tests/utils/SelfReloadingJSONRpcProvider.js:11]: Per offline conversation, let's make sure we cache the pending promise for `reloadAccount()` and chain on it if we are performing multiple parallel requests, chain on the existing pending reload request

**Review by MaximusHaximus — CHANGES_REQUESTED** [review]
:D <3 :) Comments inline

**Review by MaximusHaximus — APPROVED** [review]
This is ready to rock -- **but it looks like we've got some changes in this changeset that shouldn't be there**.

Once we've got #2299 landed, please rebase this on `master` and double-check that these are no longer showing as diffs.

File: packages/e2e-tests/playwright.config.js [hunk 2]
```diff
@@ -7,6 +7,14 @@ expect.extend(matchers);
 
 const config = {
     globalSetup: require.resolve("./global-setup.js"),
+    reporter: [["./reporters/WalletE2eLogsReporter.js", { logger: console }], ["./reporters/pagerduty-reporter.js"]],
+    webServer: {
+        command:
+            "cd ../frontend && npx serve dist -l 1234 -s --ssl-cert devServerCertificates/primary.crt --ssl-key devServerCertificates/private.pem",
```
[the remark this answers is not in the document]
Comment (esaminu) [reply packages/e2e-tests/playwright.config.js:13]: I'll do this in a follow up PR 👍

**Comment by esaminu** [comment]
> This is ready to rock -- **but it looks like we've got some changes in this changeset that shouldn't be there**.

Rebased and the changes are no longer on this changeset! 👍

**Pull request merged**